packages/client/src/StreamrClient.ts

```diff
@@ -247,41 +368,102 @@ export class StreamrClient {
         return this.streamRegistry.getStreamPublishers(streamIdOrPath)
     }
 
+    /**
+     * Gets all valid subscribers of a stream.
+     *
+     * @param streamIdOrPath - the stream id
+     * @returns async iterable collection of {@link EthereumAddress} (automatic paging)
+     */
     getStreamSubscribers(streamIdOrPath: string): AsyncIterable<EthereumAddress> {
         return this.streamRegistry.getStreamSubscribers(streamIdOrPath)
     }
 
+    /**
+     * Checks whether the given permission is in effect.
+     *
+     * @param query - defines the permission to be checked
+     * @returns resolves with true/false, rejects if the check could not be performed
+     */
     hasPermission(query: PermissionQuery): Promise<boolean> {
         return this.streamRegistry.hasPermission(query)
     }
 
+    /**
+     * Returns a list of all permissions in effect for a given stream.
+     *
+     * @param streamIdOrPath - the stream id
+     */
     getPermissions(streamIdOrPath: string): Promise<PermissionAssignment[]> {
         return this.streamRegistry.getPermissions(streamIdOrPath)
     }
 
+    /**
+     * Grants a permission on a given stream.
+     *
+     * @param streamIdOrPath - the stream id
+     * @param assignments - defines the permission(s) to be granted
+     * @returns if successful, a resolved promise
+     */
     grantPermissions(streamIdOrPath: string, ...assignments: PermissionAssignment[]): Promise<void> {
         return this.streamRegistry.grantPermissions(streamIdOrPath, ...assignments)
     }
 
+    /**
+     * Revokes a permission on a given stream.
+     *
+     * @param streamIdOrPath - the stream id
+     * @param assignments - defines the permission(s) to be revoked
+     * @returns if successful, a resolved promise
+     */
     revokePermissions(streamIdOrPath: string, ...assignments: PermissionAssignment[]): Promise<void> {
         return this.streamRegistry.revokePermissions(streamIdOrPath, ...assignments)
     }
 
+    /**
+     * Sets a list of permissions to be in effect.
+     *
+     * @remarks Can be used to set the permissions of multiple streams in one transaction. Great for doing bulk
+     * operations and thus saving gas costs. Notice that the behaviour is _set_, therefore any existing permissions not
+     * re-defined will be removed.
+     *
+     * @param items - a list of permissions to be set
+     * @returns if successful, a resolved promise
+     */
     setPermissions(...items: {
         streamId: string
         assignments: PermissionAssignment[]
     }[]): Promise<void> {
         return this.streamRegistry.setPermissions(...items)
     }
 
+    /**
+     * Checks whether a given (user) address is a valid publisher of a stream.
+     *
+     * @param streamIdOrPath - the stream id
+     * @param userAddress - the Ethereum address of the user
+     * @returns resolves with true/false, rejects if check could not be performed
```

Comment:
That is the normal `Promise` functionality, and `true`/`false` return type is obvious.